Fix issue with dependent variable update in dl_solver.py (#161)
1QB-Information-Technologies/ccvm: CHANGELOG.md

  change aims to streamline the code and eliminate unnecessary components.

> ### Changed
> - Fixed `s` update in `DLSolver._solve()` in ccvm_simulators.solvers.dl_solver.py
> - Streamlined README by relocating and optimizing architecture diagrams.

Reviewer: ```suggestion
### Fixed
- Fixed issue where `s` was being updated incorrectly on each iteration of `DLSolver._solve()`

### Changed
- Streamlined README by relocating and optimizing architecture diagrams.
```

Reviewer: Let's add a new section for bug fixes and put this one in it!

Author: I added a section and inserted the change as suggested.
